curriculum/challenges/english/20-upcoming-python/learn-string-manipulation-by-building-a-cipher/65521203d7165c7b84b22ad4.md

```diff
@@ -0,0 +1,52 @@
+---
+id: 65521203d7165c7b84b22ad4
+title: Step 18
+challengeType: 20
+dashedName: step-18
+---
+
+# --description--
+
+Repeating this process for each character in `text` can be tedious. Thankfully, you can simplify it using a loop. A loop allows you to systematically go through a sequence of elements and execute actions on each one.
+
+In this case, you'll employ a `for` loop. The syntax is as follows:
+
+```py
+for i in iterable:
+    <code>
+```
+
+`for` is the keyword denoting the loop type. `i` sequentially takes the value of the elements in `iterable`. It can have any valid name, but it's convenient to give it a meaningful name. `iterable` is the sequence you want to iterate over. Last but not least, the code to execute at each iteration constitutes the body of the loop and must be indented — typically 4 spaces — from the loop level. This indented level is a code block.
+
+Notes on indentation in Python:
+
+- Indentation is used to indicate blocks of code.
+- Statements with the same level of indentation are considered part of the same block.
+- Nested blocks are created by increasing the level of indentation.
+- A colon (:) at the end of a line is a signal that a new indented block of code will follow.
+
+Remove everything after the `alphabet` line. Then create a `for` loop to iterate over `text`. Inside the body print each character of your `text`.
```

Comment:
We need to find a way to split this info up into multiple steps.

My current suggestion is:
1) First paragraph, and spoon-feed the Camper the for-loop they should write
2) Have Camper print `i`
3) Teach about indentation by getting Camper to put `print` on wrong level - "look, no output in the console"